README.md
This repository contains very old linux distributions, kernels, packages, and their sources. These files are scattered across the internet and are difficult to find. Nevertheless, they are quite important for academic and learning purposes.

***insmod*** in this directory is 1996 version of the program. Before Linux 2.4 insmod was responsible to symbol relocation in a kernel module, while it was being installed in the kernel. Therefore this code is important to understand how that was done.

***modutils*** folder is the corresponding modutils of that time.

**Resources-**

1. http://ftp.kh.edu.tw/
1. http://ftp.kh.edu.tw/Linux/Redhat/en_6.2/
1. modutils 1996 - http://ftp.kh.edu.tw/Linux/Redhat/en_6.2/misc/src/anaconda/isys/modutils/
1. iso images - https://old-linux.com/
1. Kernels - https://www.kernel.org/pub/linux/kernel/Historic/old-versions/
1. https://www.kernel.org/pub/linux/kernel/Historic/


modutils/ChangeLog
Thu Sep 11 08:43:03 PDT 1997  Richard Henderson  <rth@cygnus.com>

	* Release version 2.1.55.

Wed Sep 10 16:34:41 PDT 1997  Richard Henderson  <rth@cygnus.com>

	* genksyms/genksyms.c (crc_prefix): New variable.
	(export_symbol): Use it.
	(main): New option -p sets it.
	* man/genksyms.8: Update documentation.
	Idea from Jacques Gelinas <jack@solucorp.qc.ca>.

Wed Sep 10 16:09:37 PDT 1997  Richard Henderson  <rth@cygnus.com>

	* insmod/Makefile: Allow insmod & rmmod to run as a combined binary.
	* insmod/main.c: New file.
	* Makeconfig (COMBINE_INSMOD_RMMOD): New define.
	Requested by Erik Troan <ewt@redhat.com>.

Wed Sep 10 15:41:16 PDT 1997  Richard Henderson  <rth@cygnus.com>

	* depmod/depmod.c (load_obj_file): Print error if fopen fails.
	(main): Continue processing on errors, so that we see all errors
	all at once.
	Reported by Gunther Mayer <gunther.mayer@braunschweig.netsurf.de>.

Tue Aug 26 11:14:36 -0400 1997  Telly Mavroidis  <mavroidi@ctc-fund.com>

	* util/sys_qm.c [__sparc__]: Fix typo.

Tue Jul 29 23:02:04 PDT 1997  Richard Henderson  <richard@gnu.ai.mit.edu>

	* include/module.h: Rename sizeof_type to tgt_sizeof_type.  Kill
	the struct duplication by defining a tgt_long.
	* insmod/insmod.c: Propogate sizeof_type change.

Tue Jul 29 22:23:16 PDT 1997  Jakub Jelinek  <jj@sunsite.mff.cuni.cz>

	* insmod/insmod.c (search_module_path): Add ipv6.

Tue Jul 29 22:23:16 PDT 1997  Jakub Jelinek  <jj@sunsite.mff.cuni.cz>

	Sparc64 support:
	* depmod/Makefile (DEFS): Pass in ARCH as well.
	* insmod/Makefile: Likewise.
	* obj/Makefile: Likewise.

	* insmod/insmod.c: Don't use native sizeof(type), use constants
	sizeof_type that correspond to the target.
	* include/module.h: Detect and cope with sparc64 target and
	sparc32 host.

	* include/obj.h: libc5 doesn't declare ELF64_ST_INFO either.

	* include/elf_sparc64.h: New file.
	* obj/obj_sparc64.c: New file.

Sat Jul 19 23:56:02 +0200 1997  Florian La Roche  <florian@knorke.saar.de>

	* depmod/misc.c (resolve_string): Kill assignment warning.
	* genksyms/genksyms.h: Extern outfile and debugfile properly.
	* man/modprobe.1: New file.

Wed Apr 16 15:27:17 +1000 1997  Paul Mackerras  <paulus@cs.anu.edu.au>

	* include/elf_ppc.h, obj/obj_ppc.c: New files.

Mon Apr 14 20:33:56 CDT 1997  Richard Henderson  <rth@tamu.edu>

	* depmod/depmod.c (print_deps_file): Fix initial array
	allocation size.

Sat Apr 12 19:23:25 CDT 1997  Richard Henderson  <rth@tamu.edu>

	* obj/obj_reloc.c (obj_relocate): If AXP_BROKEN_GAS, work
	around the old gas local symbol LITERAL reloc bug.

Mon Mar 31 23:03:12 EST 1997  David S. Miller  <davem@jenolan.rutgers.edu>

	* genksyms/lex.l (yylex) [case ST_TABLE_5]: Missing break.

Tue Mar 25 21:41:00 CST 1997  Richard Henderson  <rth@tamu.edu>

	* depmod/depmod.c (old_read_kernel_syms): Get symbols via
	get_kernel_syms syscall rather than /proc/ksyms.

	* depmod/modprobe.c (is_removable): Final arg to query_module
	is a size_t, not an int.

Sat Mar 15 19:34:07 PST 1997  H.J. Lu  <hjl@gnu.ai.mit.edu>

	* depmod/conf_file.c (release_all_sets): Set mod_set to NULL.
	(read_config_file): only call release_all_sets () once.
	call resolve_string () to expand string.

	* depmod/misc.c (resolve_string): New function that resolves
	string with `foo`.
	(what_command): New function that returns the known commands.
	(expand_command): New funtion that expands pattern or
	command. Only `uname -r` is supported this time.

	* depmod/misc.h: New prototype for resolve_string ().

	* depmod/modprobe.c (read_deps_file): call resolve_string () to
	expand string.

	* man/depmod.1: Fix typos. The syntax should be "path[TAG]=PATHNAME"
	means TAG is appended to PATHNAME.

Thu Mar 13 13:38:12 CST 1997  Richard Henderson  <rth@tamu.edu>

	* insmod/insmod.c (ncv_strcmp): New function that combines old
	m_strcmp and k_strcmp -- it's not a simple matter to always have
	the versioned symbol as the first or second argument.
	(main): Install ncv_strcmp instead.
	Reported by David Bourgin <dbourgin@wsc.com>.

	* insmod/insmod.c (get_kernel_version): Take an extra argument to
	return the full uname string.
	(old_get_module_version): Likewise.
	(new_get_module_version): Likewise.
	(main): Record and compare the full uname string, to handle
	suffixes like "-ISS" and "-SMP".

Sun Mar  9 23:52:18 CST 1997  Richard Henderson  <rth@tamu.edu>

	* genksyms/lex.l (yylex): Only use the file component of the input
	path when generating the output filename.
	Reported by Markus Dickebohm <m.dickebohm@uni-koeln.de>.

	* depmod/Makefile (clean): Remove modprobe and depmod.

Sun Mar  9 13:55:27 MET 1997  Eddie C. Dost  <ecd@skynet.be>

	* obj/obj_sparc.c (arch_apply_relocation): Handle lots
	more relocation types.

Tue Mar  4 08:12:58 CST 1997  Richard Henderson  <rth@tamu.edu>

	* Merge new depmod from Marcin Dalecki
	<dalecki@sub994.sub.uni-goettingen.de>.

	* insmod/insmod.c (get_kernel_version): Don't assert that the last
	number of the version be followed by a NUL.
	(old_get_module_version, new_get_module_version): Likewise.
	Reported by Keith Owens <kaos@ocs.com.au>

	* insmod/insmod.c: Merge character and string array patches
	from Jean Tourrilhes <jt@hplb.hpl.hp.com>.

	* kerneld/kerneld.c (kerneld_error): Add missing va_end; use vsnprintf.

Sun Feb  2 13:21:39 CST 1997  Richard Henderson  <rth@tamu.edu>

	* genksyms/lex.l (yylex): Update documentation on 2.0.x symbol table
	processing.  Generate phrases for asm blocks as well, as it is more
	efficient to do it in the lexer.
	* genksyms/parse.y (opt_asm_phrase, asm_declaration): Recognise 
	the ASM_PHRASE token rather than trying to work it out ourselves.

Sun Feb  2 07:44:00 CST 1997  Richard Henderson  <rth@tamu.edu>

	* man/kerneld.8: Format for section 8.
	* man/genksyms.8: Renamed from genksyms.1 to match how it formats.
	Reported by Martin von Loewis <martin@mira.isdn.cs.tu-berlin.de>.

Tue Jan 28 21:10:07 MET 1997  Janos Farkas  <chexum@shadow.banki.hu>


modutils/depmod/alias.h
#ident "$Id: alias.h,v 1.2 1999/09/11 15:43:57 msw Exp $"

/*
 * List of predefined aliases
 */

  {(void *) 0, "binfmt-2", "binfmt_aout"},	/* m68k */
  {(void *) 0, "binfmt-0107", "binfmt_aout"},	/* m68k */
  {(void *) 0, "binfmt-204", "binfmt_aout"},
  {(void *) 0, "binfmt-263", "binfmt_aout"},
  {(void *) 0, "binfmt-264", "binfmt_aout"},
  {(void *) 0, "binfmt-267", "binfmt_aout"},
  {(void *) 0, "binfmt-387", "binfmt_aout"},
  {(void *) 0, "binfmt-332", "iBCS"},
  {(void *) 0, "binfmt-310", "binfmt_java"},

  {(void *) 0, "block-major-1", "rd"},
  {(void *) 0, "block-major-2", "floppy"},
  {(void *) 0, "block-major-3", "ide-probe"},
  {(void *) 0, "block-major-7", "loop"},
  {(void *) 0, "block-major-8", "sd_mod"},
  {(void *) 0, "block-major-11", "sr_mod"},
  {(void *) 0, "block-major-13", "xd"},
  {(void *) 0, "block-major-15", "cdu31a"},
  {(void *) 0, "block-major-16", "gscd"},
  {(void *) 0, "block-major-17", "optcd"},
  {(void *) 0, "block-major-18", "sjcd"},
  {(void *) 0, "block-major-20", "mcdx"},
  {(void *) 0, "block-major-22", "ide-probe"},
  {(void *) 0, "block-major-23", "mcd"},
  {(void *) 0, "block-major-24", "sonycd535"},
  {(void *) 0, "block-major-25", "sbpcd"},
  {(void *) 0, "block-major-26", "sbpcd"},
  {(void *) 0, "block-major-27", "sbpcd"},
  {(void *) 0, "block-major-29", "aztcd"},
  {(void *) 0, "block-major-32", "cm206"},
  {(void *) 0, "block-major-33", "ide-probe"},
  {(void *) 0, "block-major-34", "ide-probe"},

  {(void *) 0, "char-major-4", "serial"},
  {(void *) 0, "char-major-5", "serial"},
  {(void *) 0, "char-major-6", "lp"},
  {(void *) 0, "char-major-9", "st"},
  {(void *) 0, "char-major-10", "misc"},		/* was: mouse */
  {(void *) 0, "char-major-10-0", "busmouse"},		/* /dev/logibm Logitech bus mouse */
  {(void *) 0, "char-major-10-1", "psaux"},		/* /dev/psaux PS/2-style mouse port */
  {(void *) 0, "char-major-10-2", "msbusmouse"},	/* /dev/inportbm Microsoft Inport bus mouse */
  {(void *) 0, "char-major-10-3", "atixlmouse"},	/* /dev/atibm ATI XL bus mouse */
							/* /dev/jbm J-mouse */
  {(void *) 0, "char-major-10-4", "amigamouse"},	/* /dev/amigamouse Amiga mouse (68k/Amiga) */
  {(void *) 0, "char-major-10-5", "atarimouse"},	/* /dev/atarimouse Atari mouse */
							/* /dev/sunmouse Sun mouse */
							/* /dev/beep Fancy beep device */
							/* /dev/modreq Kernel module load request */
  {(void *) 0, "char-major-10-130", "wdt"},		/* /dev/watchdog Watchdog timer port */
  {(void *) 0, "char-major-10-131", "wdt"},		/* /dev/temperature Machine internal temperature */
							/* /dev/hwtrap Hardware fault trap */
							/* /dev/exttrp External device trap */
  {(void *) 0, "char-major-10-135", "off" },		/* rtc cannot be compiled as module */
  {(void *) 0, "char-major-10-139", "openprom"},	/* /dev/openprom Linux/Sparc interface */
  {(void *) 0, "char-major-14", "sound"},
  {(void *) 0, "char-major-19", "cyclades"},
  {(void *) 0, "char-major-20", "cyclades"},
  {(void *) 0, "char-major-21", "sg"},
  {(void *) 0, "char-major-27", "ftape"},
  {(void *) 0, "char-major-34", "scc"},
  {(void *) 0, "char-major-35", "tclmidi"},
  {(void *) 0, "char-major-36", "netlink"},
  {(void *) 0, "char-major-37", "ide-tape"},
  {(void *) 0, "char-major-48", "riscom8"},
  {(void *) 0, "char-major-49", "riscom8"},
  {(void *) 0, "char-major-63", "kdebug"},

  {(void *) 0, "dos", "msdos"},
  {(void *) 0, "dummy0", "dummy"},
  {(void *) 0, "dummy1", "dummy"},
  {(void *) 0, "eth0", "off"},
  {(void *) 0, "iso9660", "isofs"},
  {(void *) 0, "md-personality-1", "linear"},
  {(void *) 0, "md-personality-2", "raid0"},
  {(void *) 0, "net-pf-1", "unix"},			/* PF_UNIX      1  Unix domain sockets */
  {(void *) 0, "net-pf-2", "ipv4"},			/* PF_INET      2  Internet IP Protocol */
  {(void *) 0, "net-pf-3", "off"},			/* PF_AX25      3  Amateur Radio AX.25 */
  {(void *) 0, "net-pf-4", "ipx"},			/* PF_IPX       4  Novell IPX */
  {(void *) 0, "net-pf-5", "appletalk"},		/* PF_APPLETALK 5  Appletalk DDP */
							/* PF_NETROM    6  Amateur radio NetROM */
							/* PF_BRIDGE    7  Multiprotocol bridge */
							/* PF_AAL5      8  Reserved for Werner's ATM */
							/* PF_X25       9  Reserved for X.25 project */
							/* PF_INET6     10 IP version 6 */
  {(void *) 0, "netalias-2", "ip_alias"},
  {(void *) 0, "plip0", "plip"},
  {(void *) 0, "plip1", "plip"},
  {(void *) 0, "ppp0", "ppp"},
  {(void *) 0, "ppp1", "ppp"},
  {(void *) 0, "scsi_hostadapter", "off"},		/* if not in config file */
  {(void *) 0, "slip0", "slip"},
  {(void *) 0, "slip1", "slip"},
  {(void *) 0, "tty-ldisc-1", "slip"},
  {(void *) 0, "tty-ldisc-3", "ppp"},
